### PR TITLE
Add documentation for how to generate MAPI TS types and client functions

### PR DESCRIPTION
**Pull request by kuosandys**
### What does this PR do?

This PR adds documentation for how to generate TS types and client functions to FE docs, as implemented in https://github.com/giantswarm/happa/pull/3822 and https://github.com/giantswarm/happa/pull/3843.

### Any background context you can provide?

As a last step to https://github.com/giantswarm/roadmap/issues/1528. Depends on the changes from https://github.com/giantswarm/happa/pull/3843.

### Do the docs need to be updated?

Yes we're doing it with this PR 🙃

### CODE REVIEW

File: src/stories/GeneratingMapiTypesClientFunctions.stories.mdx [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+
+# Generating TS types and client functions for the MAPI client
+
+## Introduction
+happa uses an HTTP client to communicate with the management cluster's Kubernetes API (MAPI). In order to facilitate making requests to the MAPI and working with MAPI custom resources, we define client functions and TypeScript (TS) types. These are generated from the resources' Custom Resource Definitions (CRDs).
+
+## How to generate
+Run `yarn generate-mapi` from happa's base directory.
+
+This generates TS types and MAPI client functions for resources declared in [`scripts/generate/mapi-resources.yaml`](`https://github.com/giantswarm/happa/blob/main/scripts/generate/mapi-resources.yaml`), and formats the generated files with Prettier.
```
Comment (mogottsch) [inline src/stories/GeneratingMapiTypesClientFunctions.stories.mdx:13]: It's `scripts/generateTypes/mapi-resources.yaml`, right?

File: src/stories/GeneratingMapiTypesClientFunctions.stories.mdx [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+
+
+## Declaring resources
+Resources for which we wish to generate TS types and client functions can be declared in [`scripts/generate/mapi-resources.yaml`](`https://github.com/giantswarm/happa/blob/main/scripts/generate/mapi-resources.yaml`).
```
Comment (mogottsch) [inline src/stories/GeneratingMapiTypesClientFunctions.stories.mdx:17]: same here

File: src/stories/GeneratingMapiTypesClientFunctions.stories.mdx [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+`name` is the name of the resource, and should be given in PascalCase. This value, prepended with an `I`, will be used as the name for the generated TS interface. To reduce confusion, this should preferably match the `.spec.names.kind` field of the resource's CRD.
+
+##### crdURL
+`crdURL` is the URL at which the `.yaml` file fo the CRD can be found. If referencing a CRD from GitHub, please make sure to use the URL of the raw file contents.
+
+##### verbs
+`verbs` specifies the Kubernetes verbs we wish to use to interact with a resource. Client functions will only be generated for the defined verbs. Value should be one of: `get`, `list`, `update`, `create`, or `delete`.
+
+## Providing custom TS types
+For most cases it should not be necessary to overwrite the types defined by the schema defined in the resource's CRD. However, to allow the generated TS types to work better with happa's codebase, custom TS types can be provided via the `getTSTypesConfig` method in [`scripts/generate/getTypes`](https://github.com/giantswarm/happa/blob/main/scripts/generate/getTypes.ts).
```
Comment (mogottsch) [inline src/stories/GeneratingMapiTypesClientFunctions.stories.mdx:50]: same here

**Review by mogottsch — CHANGES_REQUESTED** [review]
Clear documentation :+1: 
I think `scripts/generate/mapi-resources.yaml` should be `scripts/generateTypes/mapi-resources.yaml`, right?

**Comment by kuosandys** [comment]
> I think `scripts/generate/mapi-resources.yaml` should be `scripts/generateTypes/mapi-resources.yaml`, right?

The folder was renamed with https://github.com/giantswarm/happa/pull/3843 since the code in the folder wasn't just about generating types anymore 😅 I should have clarified in the PR description that this docs change depends on the changes in that pull request - thanks for pointing it out! I've updated the branch and the description above 🙂

File: src/stories/GeneratingMapiTypesClientFunctions.stories.mdx [hunk 4]
```diff
@@ -0,0 +1,59 @@
...
+
+### Example
+Below is an example definition for a resource group entry:
+
+```yaml
+- apiVersionAlias: capgv1beta1
+  apiVersion: infrastructure.cluster.x-k8s.io/v1beta1
+  resources:
+  - name: GCPCluster
+    crdURL: https://raw.githubusercontent.com/kubernetes-sigs/cluster-api-provider-gcp/main/config/crd/bases/infrastructure.cluster.x-k8s.io_gcpclusters.yaml
```
Comment (gusevda) [inline src/stories/GeneratingMapiTypesClientFunctions.stories.mdx:27]: should we add `verbs` here?

Comment (kuosandys) [reply src/stories/GeneratingMapiTypesClientFunctions.stories.mdx:27]: Yes, we should 😄

**Review by gusevda — APPROVED** [review]
Looks good!

**Review by mogottsch — APPROVED** [review]

**Pull request merged**